### PR TITLE
Add [Bugzilla] bugstatus badge

### PR DESCRIPTION
**Pull request by eliperelman**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -7814,6 +7814,58 @@ cache(function(data, match, sendBadge, request) {
...
+  var options = {
+    method: 'GET',
+    json: true,
+    uri: 'https://bugzilla.mozilla.org/rest/bug/' + bugNumber
+  };
+  var badgeData = getBadgeData('bugstatus', data);
+  request(options, function (err, res, json) {
+    badgeData.text[0] = 'Bug ' + bugNumber;
+
+    if (err !== null) {
```
Comment (RedSparr0w) [inline server.js]: Could you change this to use [`checkErrorResponse()`](https://github.com/badges/shields/blob/master/server.js#L552-L555) here, as it helps to lessen code duplication while still covering most cases.

**Comment by RedSparr0w** [comment]
Thanks for this PR,

Though I think it may look a little nicer in lowercase and keep with the consistency of the other badges.

File: server.js (hunk 1 above)
Comment (eliperelman) [reply server.js]: Sure thing.

**Comment by eliperelman** [comment]
@RedSparr0w I'm kinda torn on that. While I agree it looks better, these values map directly to what users see in the BZ UI:

screen shot 2018-02-08 at 6 32 46 pm
screen shot 2018-02-08 at 6 32 53 pm

That said, if you still would like it changed, let me know and I will. Thanks!

**Comment by RedSparr0w** [comment]
Good point, I can definitely see where you are coming from, though i think the all lowercase will better match the current badges.


Also what do you think about removing `resolved` from the badges?

**Comment by eliperelman** [comment]
I like the last column!

File: server.js [hunk 2]
```diff
@@ -7888,6 +7888,73 @@ camp.route(/^\/vaadin-directory\/(star|status|rating|rc|rating-count|v|version|r
...
+  var format = match[2];
+  var options = {
+    method: 'GET',
+    json: true,
+    uri: 'https://bugzilla.mozilla.org/rest/bug/' + bugNumber
+  };
+  var badgeData = getBadgeData('Bug ' + bugNumber, data);
+  request(options, function (err, res, json) {
+    if (checkErrorResponse(badgeData, err, res)) {
+      badgeData.text[1] = 'unknown';
```
Comment (RedSparr0w) [inline server.js]: could you remove this line, as [`checkErrorResponse()`](https://github.com/badges/shields/blob/3cb71037fdbed20f74a18880ed9edad4273a08c5/lib/error-helper.js#L3) should set `badgeData.text[1]` to a relevant message.

File: server.js [hunk 3]
```diff
@@ -7888,6 +7888,73 @@ camp.route(/^\/vaadin-directory\/(star|status|rating|rc|rating-count|v|version|r
...
+camp.route(/^\/bugzilla\/(\d+)\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var bugNumber = match[1];  // eg, 1436739
+  var format = match[2];
+  var options = {
+    method: 'GET',
+    json: true,
+    uri: 'https://bugzilla.mozilla.org/rest/bug/' + bugNumber
+  };
+  var badgeData = getBadgeData('Bug ' + bugNumber, data);
```
Comment (RedSparr0w) [inline server.js]: could you also change this to be lower case
eg:

File: service-tests/bugzilla.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  }));
+
+t.create('Bugzilla invalid bug status')
+  .get('/83548978974387943879.json')
+  .expectJSON({ name: 'Bug 83548978974387943879', value: 'unknown' });
+
+t.create('Bugzilla failed request bug status')
+  .get('/996038.json')
+  .networkOff()
+  .expectJSON({ name: 'Bug 996038', value: 'unknown' });
```
Comment (RedSparr0w) [inline service-tests/bugzilla.js]: could you also add 1 more test for an invalid JSON response.
[example](https://github.com/badges/shields/blob/master/service-tests/cdnjs.js#L28-L34)

**Comment by paulmelnikow** [comment]
Hi @eliperelman would you be able to respond to the last round of comments? Would be great to get this merged!

File: service-tests/bugzilla.js (hunk 4 above)
Comment (eliperelman) [reply service-tests/bugzilla.js]: I made all the changes except this one, for some reason I couldn't get this to work as nock kept throwing `NetConnectNotAllowedError` errors.

**Comment by RedSparr0w** [comment]
Thanks for the changes,
All looks good to me.

Merged!
😄

**Comment by eliperelman** [comment]
@RedSparr0w thanks! Any idea when this lands on the production site?

**Comment by paulmelnikow** [comment]
See #1538

**Pull request merged**